client_test.go

```diff
@@ -38,3 +39,13 @@ func Test_Client_Clone(t *testing.T) {
 	assert.Equal(t, "test-token", clone.clientRequestModifiers.headers.token)
 	assert.Equal(t, "test-namespace", clone.clientRequestModifiers.headers.namespace)
 }
+
+// assertEqual compares the two clients, accounting for nested func pointers
```

Comment:
doesn't `testify` do this already?

Reply:
Unfortunately not,

```go
f := func() {
}
assert.Equal(t, f, f)
```

fails with

```
Invalid operation: (func())(0x1002a1300) == (func())(0x1002a1300) (cannot take func type as argument)
```

Similarly, `reflect.DeepEqual` also fails: https://go.dev/play/p/bGU_d8CM2xg

Reply:
ahh gotcha, can you update the comment adding `does not work with testify` (or something similar) please?

Reply:
https://github.com/hashicorp/vault-client-go/pull/231/commits/5ba77bbc0559c703e40be74c99f0bce354488648